app/models/category.py

```diff
@@ -1,102 +1,122 @@
-from sqlalchemy import (
-    String,
-    Boolean,
-    Integer,
-    Text,
-    Date,
-    DECIMAL,
-    ForeignKey,
-    Enum,
-)
+from sqlalchemy import String, Boolean, Text, Date, DECIMAL, ForeignKey
 from sqlalchemy.orm import Mapped, mapped_column, relationship
 from uuid import UUID
-from typing import List
+from typing import List, Optional
 from .base import Base
-from .user import User
 from app.enum.category import CategoryNameEnum, CategoryColorEnum
 
 
-# CategoryProgress 테이블 (한 카테고리에 여러 목표, 노트가 연결됨)
-class CategoryProgress(Base):
-    __tablename__ = "category_progresses"
+# ✅ Category 테이블: 사용자의 카테고리를 정의 (코딩/영어/운동 등)
+class Category(Base):
+    __tablename__ = "categories"
 
-    user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), nullable=False)
-    category_name: Mapped[CategoryNameEnum] = mapped_column(
-        Enum(CategoryNameEnum, name="category_name_enums"), nullable=False
-    )
-
-    total_goal: Mapped[int] = mapped_column(Integer, nullable=False)
-    completed_goal: Mapped[int] = mapped_column(Integer, nullable=False)
-    progress_rate: Mapped[DECIMAL] = mapped_column(DECIMAL(5, 2), default=0.00)
-
-    # CategoryProgress → User (N : 1 관계)
-    user: Mapped["User"] = relationship("User", back_populates="category_progresses")
+    category_name: Mapped[CategoryNameEnum] = mapped_column(nullable=False)
+    category_color: Mapped[CategoryColorEnum] = mapped_column(nullable=False)
 
-    # CategoryProgress → ComprehensiveEvaluation, RecommendedChallenge, Goal, MonthlyAchievement (1 : N 관계)
-    comprehensive_evaluations: Mapped[
-        List["app.models.evaluation.ComprehensiveEvaluation"]
-    ] = relationship(
-        "app.models.evaluation.ComprehensiveEvaluation",
-        back_populates="category_progress",
+    # 관계: Category 1 : 1 Progress
+    progress: Mapped["app.models.category.Progress"] = relationship(
```

Comment:
ERD에서는 categories와 progresses가 1대N인데 ERD가 잘못 작성된걸까요?

그리고 여기는 `Optional`이 없어도 되는건지 궁금합니당. @trivialcoding6 

```suggestion
    # 관계: Category 1 : 1 Progress
    progress: Mapped[Optional["app.models.category.Progress"]] = relationship(
```

Reply:
@GideokKim 
네!! 카테고리와 진행률 테이블은 1:1이 맞습니다! 1:1 관계이기 때문에 카테고리가 있으면 그에 대응하는 progress도 있어서 필수값으로 생각했습니다

cc @Ryan-OH @chaeengg @codingorijazz

Reply:
지금 채은님 erd가 완성본이 아니여서 erd최신화 되고 나서 erd와 비교하시는게 좋을것 같아요!